analysis_driver/pipelines/rapid.py

```diff
@@ -0,0 +1,238 @@
+                                '1-2': float(map_metrics['PCT of genome with coverage [ 1x: 2x)'][0]),
+                                '0-1': float(map_metrics['PCT of genome with coverage [ 0x: 1x)'][0]),
+                            }
+                        }
+                    }
+                }
+            )
+
+        rest_communication.post_or_patch('samples', data, id_field='sample_id')
+        return 0
+
+
+class DragenOutput(RapidStage):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._datestamp = None
+
+    @property
+    def datestamp(self):
+        if self._datestamp is None:
+            self._datestamp = date.today().isoformat()
+        return self._datestamp
+
+    def delivery_folder(self, sample):
+        return os.path.join(
+            cfg['delivery']['dest'],
+            sample.project.name,
+            self.datestamp + '_rapid',
+            sample.udf['User Sample Name']
+        )
+
+    def _run(self):
+        exit_status = 0
+
+        for lane in sorted(self.dataset.rapid_samples_by_lane):
+            sample = self.dataset.rapid_samples_by_lane[lane]
+            sample_id = sample['sample_id']
+
+            if not self.review(sample_id):
+                continue
+
+            output_dir = os.path.join(cfg['sample']['output_dir'], sample['project_id'], sample_id, 'rapid_analysis')
+            output_status = self.output_data(lane, sample, output_dir)
+            if output_status != 0:
+                self.warning('Data output failed for sample %s', sample_id)
+                continue
+
+            exit_status += self.deliver_data(sample_id, sample['User Sample Name'], sample['project_id'], output_dir)
+
+        return exit_status
+
+    def review(self, sample_id):
+        rest_communication.post_entry(
+            'actions',
+            {'action_type': 'automatic_rapid_review', 'sample_id': sample_id},
+            use_data=True
+        )
+
+        rapid_metrics = rest_communication.get_document('samples', where={'sample_id': sample_id})['rapid_analysis']
+        if rapid_metrics['reviewed'] == 'pass':
+            return True
+        else:
+            self.warning(
+                'Sample %s has status %s: %s',
+                sample_id,
+                rapid_metrics['reviewed'],
+                rapid_metrics.get('review_comments', '(no comments)')
+            )
+            return False
+
+    def output_data(self, lane, sample, output_dir):
+        user_sample_id = sample.udf['User Sample Name']
+        staging_dir = os.path.join(self.job_dir, 'linked_output_files_%s' % lane)
+
+        os.makedirs(staging_dir, exist_ok=True)
+        transfer_data.create_output_links(
+            self.job_dir,
+            'rapid_analysis',
+            staging_dir,
+            user_sample_id=user_sample_id,
+            lane=lane
+        )
+
+        # TODO: implement generic MD5Sum stage
```

Comment:
Should this TODO be here, or is it better suited for a GitHub issue?

Reply:
See #274

Reply:
Great, maybe we can discuss avoiding TODOs in the source code with @tcezard just to keep things tidy, or else including the issue number so there's better traceability? It's up for discussion.

Reply:
Can you add a reference to #274 in the TODO